### PR TITLE
tests: move spread-shellcheck to run-checks --static

### PR DESCRIPTION
**Pull request by chipaca**
spread-shellcheck is fast enough we can run it as part of the static
suite on travis.

### CODE REVIEW

File: run-checks [hunk 1]
```diff
@@ -185,6 +185,8 @@ if [ "$STATIC" = 1 ]; then
             exit 1
         fi
         unset regexp
+        echo Checking shell scripts inside spread yamls...
+        ./spread-shellcheck spread.yaml tests
```
Comment (bboozzoo) [inline run-checks]: The spread task one used the latest version of shellcheck from snaps. Any chance we could use something more recent(-ish) here too?

Comment (mvo5) [reply run-checks]: Maybe we could keep the shellcheck spread in addition to the fast check?

Comment (bboozzoo) [reply run-checks]: If Travis supports snaps (does it?) we could `snap install` before :)

**Comment by chipaca** [comment]
@bboozzoo @mvo5 with this test adding a whole minute to the 'quick' run, I'm not sure we want it in any case

**Comment by chipaca** [comment]
@mvo @bboozzoo I changed the approach a little bit. With the current rearrangement, the 'quick' stage could be about a minute faster when things are good, or two minutes slower when we're hammering travis. It's also that little bit cleaner imho.
@bboozzoo note I tweaked `spread-shellcheck` a bit more.

File: spread-shellcheck [hunk 2]
```diff
@@ -237,8 +234,19 @@ def main(opts):
 
 
 if __name__ == '__main__':
+    # fiddle with PATH so we pick up a snapped shellcheck first
+    if "/snap/bin" in os.environ["PATH"]:
```
Comment (bboozzoo) [inline spread-shellcheck]: It's probably much easier to drop the piece that mangles the PATH and have this instead:
```
SHELLCHECK = os.getenv('SHELLCHECK', 'shellcheck)
...
def checksection(data):
    ...
    subprocess.Popen([SHELLCHECK, "-s", "bash", "-x", "-"]),
                     stdout=subprocess.PIPE,
                     stdin=subprocess.PIPE)
    ...
```

Then `SHELLCHECK=/snap/bin/shellcheck ./spread-shellcheck ...` should call `shellcheck` from snap.

File: spread-shellcheck [hunk 3]
```diff
@@ -237,8 +234,19 @@ def main(opts):
...
+        for e in os.environ["PATH"].split(":"):
+            if "/snap/bin" in e:
+                path.insert(0, e)
+            else:
+                path.append(e)
+        os.environ["PATH"] = ":".join(path)
+
     opts = parse_arguments()
     if opts.verbose or D or V:
+        subprocess.call(["shellcheck", "--version"])
```
Comment (bboozzoo) [inline spread-shellcheck]: This one then becomes `subprocess.call([SHELLCHECK, "--version"])`

**Pull request closed without merging**